src/Formats/ProtobufSerializer.cpp

```diff
@@ -170,7 +169,7 @@ namespace
         }
         if (field_descriptor.options().has_packed())
             return field_descriptor.options().packed();
-        return field_descriptor.file()->syntax() == google::protobuf::FileDescriptor::SYNTAX_PROTO3;
+        return field_descriptor.is_packed();
```

Comment:
L. 170-172 look weird. Can't we just say

```cpp
return field_descriptor.is_packed();
```

? If I read the API docs of `is_packed()` right, it works for v2 and v3 syntax and editions:


```
// Whether or not this field is packable and packed.  In proto2, packable
// fields must have `packed = true` specified.  In proto3, all packable fields
// are packed by default unless `packed = false` is specified.
bool is_packed() const;
```

Reply:
You're right, they can be dropped. I'll remove them in a follow-up commit.